### PR TITLE
docs: fix model sharing file names

### PR DESCRIPTION
**Pull request by borisdayma**
Documentation references `vocab.txt` while I think it should be `vocab.json`.

Additional notes:
* I also have `training_args.bin` and `merges.txt` which I understand are not needed to be uploaded
* I didn't get `added_tokens.json` but I understand it's not always present

### CODE REVIEW

File: docs/source/model_sharing.rst [hunk 1]
```diff
@@ -112,7 +112,7 @@ Make sure there are no garbage files in the directory you'll upload. It should o
 - a `tf_model.h5` file, which is the TensorFlow checkpoint (unless you can't have it for some reason) ;
 - a `special_tokens_map.json`, which is part of your :doc:`tokenizer <main_classes/tokenizer>` save;
 - a `tokenizer_config.json`, which is part of your :doc:`tokenizer <main_classes/tokenizer>` save;
-- a `vocab.txt`, which is the vocabulary of your tokenizer, part of your :doc:`tokenizer <main_classes/tokenizer>`
+- a `vocab.json`, which is the vocabulary of your tokenizer, part of your :doc:`tokenizer <main_classes/tokenizer>`
```
Comment (julien-c) [inline docs/source/model_sharing.rst:115]:
```suggestion
- files named `vocab.json`, `merges.txt`, `vocab.txt`, or similar, which contain the vocabulary of your tokenizer, part of your :doc:`tokenizer <main_classes/tokenizer>`

```

**Comment by julien-c** [comment]
It's more general than vocab.txt or vocab.json so I'll let @sgugger chime in as I think he wrote those lines.

**Comment by sgugger** [comment]
I didn't know that when I wrote it (and the file was vocab.json for me ;-) ). This looks good to me with the suggestion, thanks for clarifying!

**Comment by borisdayma** [comment]
Looks good on my side, the only file removed from my example would be `training_args.bin` which makes sense.

**Comment by borisdayma** [comment]
Let me know if I need to do any other changes

**Comment by sgugger** [comment]
Oops, sorry this got lost in my notifications. Don't hesitate to ping me if it happens again @borisdayma, will merge as soon as the CI is green.

**Pull request merged**